### PR TITLE
fix test_cameracalibration bug

### PR DESCRIPTION
**Pull request by NerdToMars**
fix the rotation matrix transpose issue, resolve the hug difference between perViewErrors and goodPerviewErrors via:
1. remove rotation matrix transpose in calibration test
2. add rotation matrix transpose to match good rotation matrix.

for details, please refer to this issue: https://github.com/opencv/opencv/issues/19849

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by asenyaev** [comment]
jenkins cn please retry a build

**Comment by asenyaev** [comment]
jenkins cn please retry a build

**Comment by asenyaev** [comment]
jenkins cn please retry a build

**Comment by asmorkalov** [comment]
@NerdToMars Thanks for the patch! @vpisarev is very busy now.
- The change affects `project` call results. In case, if it reduces difference between `perViewErrors` and `goodPerviewErrors` than it's logical to reduce threshold at line: `code = compare(&perViewErrors[0], &goodPerViewErrors[0], numImages, 1.1, "per view errors vector");`

**Comment by asmorkalov** [comment]
Also the same issue exists in 3.4 branch. OpenCV team merges 3.4 to master on weekly or be-weekly bases, so all changes will appear in master too. The PR was created from master branch and cannot be re-targeted to 3.4 easily. Please create PR with fix from topic brianch derived from 3.4 and point it to 3.4..

File: modules/calib3d/test/test_cameracalibration.cpp [hunk 1]
```diff
@@ -812,7 +813,6 @@ void CV_CameraCalibrationTest_CPP::calibrate(Size imageSize,
     {
         Mat r9;
         cvtest::Rodrigues( rvecs[i], r9 );
-        cv::transpose(r9, r9);
```
Comment (alalek) [inline modules/calib3d/test/test_cameracalibration.cpp:815]: added by @vpisarev here #13081

https://github.com/opencv/opencv/pull/13081/files#diff-3122fe7397a59cc808249e6c0ac6e1a089997df71021aade6778799f59b925f7R815

There is not such line on 3.4 branch.

**Comment by asmorkalov** [comment]
@NerdToMars Friendly reminder.

**Comment by asmorkalov** [comment]
@NerdToMars Friendly reminder.

**Comment by asmorkalov** [comment]
Replaced by https://github.com/opencv/opencv/pull/20810

**Pull request closed without merging**